src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -264,9 +264,12 @@ void DatasetManager::HandleSet(Coap::Header &aHeader, Message &aMessage, const I
             aMessage.Read(offset + sizeof(Tlv), sizeof(timestamp), &timestamp);
         }
 
-        // verify the request does not include fields that affect connectivity
-        if (tlvType == Tlv::kChannel || tlvType == Tlv::kMeshLocalPrefix ||
-            tlvType == Tlv::kPanId || tlvType == Tlv::kNetworkMasterKey)
+        // verify the request only includes dataset tlvs that do not affect connectivity
+        if (tlvType != Tlv::kActiveTimestamp && tlvType != Tlv::kPendingTimestamp &&
+            tlvType != Tlv::kChannelMask && tlvType != Tlv::kExtendedPanId &&
+            tlvType != Tlv::kNetworkName && tlvType != Tlv::kPSKc &&
+            tlvType != Tlv::kSecurityPolicy && tlvType != Tlv::kDelayTimer &&
+            tlvType != Tlv::kCommissionerSessionId)
```

Comment:
I do not think that it checks.
The check here only wants to check MGMT_ACTIVE_SET/MGMT_PENDING_SET only includes TLVs listed in spec, then mTlvs only includes operational dataset TLVs.